### ISSUE TITLE
release new version but remove some packages

### ISSUE THREAD

**Issue by dirk-thomas** [ros-infrastructure/bloom#425]
Consider the following scenario:

* repo `R` with packages `foo` and `bar`
* release version 1 in the upstream repo
* run `bloom-release` (all good)
* remove package `foo` from repo `R`
* release version 2 in the upstream repo
* run `bloom-release`...

In this case the second `bloom-release` call will run the generator for package `bar` version 2 as expected. But it additionally runs the generator for package `foo` version 1. While that doesn't do any harm it is unnecessary, slows down the generation, and might confuse the user.

**Comment by sloretz** [comment]
I think I'm a confused user who is on "Release version 3" of `bar`. After the package generation I see a `git push --tags` fail to push tags for `foo` version 1 because they already exist in the release repo. Do you think the reason the tags already exist is they were pushed after the generator for `foo` version 1 ran while releasing version 2 of `bar`?

**Comment by mikaelarguedas** [comment]
Yes it could be, I _think_ that in that case bloom suggests you to push --force the tags to replace the existing one

**Comment by SteveMacenski** [comment]
Just noting this as a "hey I ran into this today". The documentation in this ticket was sufficient for me to resolve - thanks for the paper trail! I'm not sure if this is something folks often run into, if so I can put it in my icebox to look at.
